src/main/java/com/pnu/dev/radioserviceapi/service/VideoServiceImpl.java

```diff
@@ -1,7 +1,168 @@
+        } else {
+
+            videoFromYoutube = videoFromYoutube
+                    .toBuilder()
+                    .priority(1)
+                    .build();
+
+            List<Video> videosToIncrementPriority = videoRepository
+                    .findAll();
+
+            videoRepository.save(videoFromYoutube);
+            incrementPriority(videosToIncrementPriority);
+        }
+    }
+
+    private Video findVideoOnYoutube(String link) {
+
+        String id = YoutubeVideoIdExtractor.getVideoIdFromLink(link).orElseThrow(() -> {
+            throw new RadioServiceAdminException("Посилання не відповідає Youtube посиланню");
+        });
+
+        YoutubeApiResult<ItemYoutubeVideosResponse> apiResult = youtubeApiClient.findVideo(id);
+
+        if (apiResult.isError()) {
+            throw new RadioServiceAdminException(apiResult.getErrorMessage());
+        }
+
+        ItemYoutubeVideosResponse itemYoutubeVideosResponse = apiResult.getData();
+        return videoMapper.itemVideosResponseToVideo(itemYoutubeVideosResponse);
+    }
+
+    @Override
+    @Transactional
+    public void updatePriority(String id, Integer newPriority) {
+
+        long numberOfVideos = videoRepository.count();
+
+        if (newPriority < 1 || newPriority > numberOfVideos) {
+            throw new RadioServiceAdminException(
+                    String.format("Приорітет повинен бути в межах від 1 до %s (кількість всіх відео)", numberOfVideos));
+        }
+        Video video = findVideoByIdOrThrowException(id);
+        updatePriority(video, newPriority);
+    }
+
+    private void updatePriority(Video videoFromDb, Integer newPriority) {
+
+        int previousPriority = videoFromDb.getPriority();
+
+        if (previousPriority == newPriority) {
+            return;
+        }
+
+        Video updatedVideo = videoFromDb
+                .toBuilder()
+                .priority(newPriority)
+                .build();
+
+
+        if (previousPriority > newPriority) {
+            List<Video> videosToIncrementPriority = videoRepository
+                    .findAllByPriorityBetween(newPriority - 1, previousPriority);
+            incrementPriority(videosToIncrementPriority);
+        }
+        if (previousPriority < newPriority) {
+            List<Video> videosToDecrementPriority = videoRepository
+                    .findAllByPriorityBetween(previousPriority, newPriority + 1);
+            decrementPriority(videosToDecrementPriority);
+        }
+
+        videoRepository.save(updatedVideo);
+    }
+
+    private Video findVideoByIdOrThrowException(String id) {
+
+        return videoRepository.findById(id).orElseThrow(() -> new RadioServiceAdminException("Відео не знайдено"));
+    }
+
+    private void incrementPriority(List<Video> videoListToIncrementPriority) {
+
+        List<Video> updatedVideos = videoListToIncrementPriority.stream()
+                .map(v -> {
+                    v.setPriority(v.getPriority() + 1);
```

Comment:
I got why of was `peek` before.
Do not use setter here, use `toBuilder` and create new object out of old, like you do above, when update priority for single video.